### DISCUSSION TITLE
Manga Support?

### DISCUSSION THREAD

**Discussion by vengeance101** [magnuschase293/comic_getter#24] [Ideas]
Magnus, this is great work! It's so neat to see our tiny discussion over at the other repo turn into an actual working tool! I hope Selenium was simple to work with, looks a lot easier than I thought. 

It looks like Xonshiz's comic-dl is now also broken 100% - it no longer works for manga either. What would it take to incorporate a Manga website in here as well? I assume a different selenium script for each website?

**Comment by magnuschase293** [comment]
Thanks, it means a lot. Sincerely I've been quiet occupied lately since I started studying. I still owe Peepijeep the windows exe and hope to add a method of tracking the downloads when running chrome in headless mode.

 To add manga support would require a custom script for every webpage. Maybe I can try and add support for one... Any  suggestions??? It will take some time, but it is an interesting project and the core ideas and functions should work with a few changes. Besides selenium is just amazing.

**Comment by vengeance101** [comment]
I think MangaEden is the main one that Xonshiz used. They have an API, but not entirely sure it's worth interfacing with versus just a selenium scrape. I'm envisioning just having an -m tag if it's manga, and then routing to the mangaeden-specific scrape.

**Comment by magnuschase293** [comment]
Is it only me or mangaeden is very unstable. 6 out of 10 mangas I tried to open had a 404 not found error??

**Comment by vengeance101** [comment]
Wow, that's a high %. I'll admit I haven't had any issues with it, but I've always just used Xonshiz's tool.

**Comment by vengeance101** [comment]
I did notice that it seems to only have single-page views, so a selenium scrape would need to hit a url for every page to get the image link.

**Comment by magnuschase293** [comment]
It is far easier than I expected the links of each chapter and page are constructed following this guidelines:

* corelink/collection/chapter number/page number

So i can simply iterate through them with a for loop and a range.

Also it is not cloudflare protected so there is no need for Selenium and I can simply use Requests to scrap the link to the actual .jpg file. 

Whenever I find some free time I will write the script, but it will be much less complex than RCO.

**Comment by vengeance101** [comment]
I did catch that One Piece is 404'd. I wonder if it's a copyright thing, or if it's just down.

**Comment by vengeance101** [comment]
Looks like MangaEden's API stopped being maintained about 2 months ago:

https://www.reddit.com/r/MangaEden/comments/hf184d/api_not_working/

Maybe it's better to develop for a different site? I am not sure which one would be best though.

**Comment by magnuschase293** [comment]
[MangaEdex](https://mangadex.org/) seems to be a good alternative.

**Comment by StrangeDOS** [comment]
I also used  Xonshiz's comic-dll.  It was a good tool for its time, but comic_getter is quickly shaping up to be even better.

Love the name too.  Comic_Getter...it gets comics.  It just makes me smile.

**Comment by vengeance101** [comment]
I really appreciate the documentation here so far. Xonshiz's tool really was not friendly when it came to jumping into it and understanding what was happening. This has a good design so far and is well documented.

**Comment by magnuschase293** [comment]
I have some extra time so I might be rturning to this idea. Do you still want anime support??? Any website in mind???

**Reply by vengeance101** [reply]
Hey! Seeing this now after a long break away from some comics. I think manga support would be GREAT, but I'm not entirely sure what websites are best for it.

**Reply by magnuschase293** [reply]
Sincerely, I haven't read any manga so I have no idea which pages to look for. Besides almost every website is an entirely different universe. So just leave the link to the one with biggest catalogue and I will see what I can do.

**Reply by vengeance101** [reply]
I'll poke around for a bit. I also don't read a lot of manga but have been getting into it more. We'll see if we can find a decent site.
